src/hooks/usePollDatasetStatus.js

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState, useRef } from 'react'
+import { useDatasetManager } from 'hooks/useDatasetManager'
+
+export const usePollDatasetStatus = () => {
+  const {
+    datasetAccessions,
+    setDatasetAccessions,
+    processingDatasets,
+    setProcessingDatasets,
+    getDataset
+  } = useDatasetManager()
+  const [polledDatasetId, setPolledDatasetId] = useState(null)
+  const [polledDatasetState, setPolledDatasetState] = useState(false)
+  const timerRef = useRef(null)
+  const isProcessingDataset = polledDatasetState?.is_processing
+
+  // polls the latest state of the processing dataset per minute
+  // (the processing usually takes a few minutes)
+  useEffect(() => {
+    if (polledDatasetId) {
+      timerRef.current = setInterval(() => {
+        refreshProcessingDataset()
+      }, 1000 * 60)
+    }
+
+    return () => clearInterval(timerRef.current)
+  }, [polledDatasetId])
+
+  // if one-off, sets the matched accession code's dataset ID to polledDatasetId
+  const pollDatasetAccession = (accessionCode) => {
+    if (Object.hasOwn(datasetAccessions, accessionCode)) {
+      pollDatasetId(datasetAccessions[accessionCode])
+    }
+  }
+
+  // sets the mached dataset ID to polledDatasetId
+  const pollDatasetId = (datasetId) => {
+    if (processingDatasets.includes(datasetId)) setPolledDatasetId(datasetId)
```

Comment:
If you also update `isProcessing` to true when the dataset id is in `processingDatasets`, then the exported `isProcessing` will trigger a render and display correctly for the time before the first interval is expired.

Reply:
The new state  was added at a5bf82650a3adc2e4925149acbcb0b3b4820ee21.

Reply:
See update at a5bf82650a3adc2e4925149acbcb0b3b4820ee21.